extension/js/common/helpers.ts

```diff
@@ -63,38 +71,49 @@ const parseAndCheckPrivateKeys = async (decryptedPrivateKeys: string[]) => {
 };
 
 const filterKeysToSave = async (candidateKeys: Key[], existingKeys: KeyInfoWithIdentity[]) => {
+  // todo: check for uniqueness of candidateKeys identities here?
   if (!existingKeys.length) {
-    return candidateKeys;
+    return { keysToRetain: [], unencryptedKeysToSave: candidateKeys };
   }
-  const result: Key[] = [];
+  const keysToRetain: Key[] = [];
+  const unencryptedKeysToSave: Key[] = [];
   for (const candidate of candidateKeys) {
     const longid = KeyUtil.getPrimaryLongid(candidate);
     const keyToUpdate = existingKeys.filter(ki => ki.longid === longid && ki.family === candidate.family);
     if (keyToUpdate.length === 1) {
       const oldKey = await KeyUtil.parse(keyToUpdate[0].private);
```

Comment:
Let's call this `parsedExistingKey`. We don't know if it's `old` or not (with respect to lastModified) so it can get confusing.